plugins/org.jkiss.dbeaver.ui.app.standalone/src/org/jkiss/dbeaver/ui/app/standalone/update/VersionUpdateDialog.java

```diff
@@ -253,104 +251,107 @@ protected void buttonPressed(int buttonId)
-                            log.debug("Downloading installation file to " + file);
-                            WebUtils.downloadRemoteFile(monitor, "Obtaining installer", getDownloadURL(app, installer, newVersion), file, null);
-                        } catch (IOException e) {
-                            return GeneralUtils.makeErrorStatus(CoreMessages.dialog_version_update_downloader_error_cannot_download, e);
-                        } catch (InterruptedException e) {
-                            log.debug("Canceled by user", e);
-                            return Status.OK_STATUS;
+                            installer.run(file, log);
+                        } catch (Exception e) {
+                            log.error("Failed to run the installer script", e);
                         }
+                    }
+                };
 
-                        if (UIUtils.confirmAction(CoreMessages.dialog_version_update_downloader_title, NLS.bind(CoreMessages.dialog_version_update_downloader_confirm_install, app.getName()))) {
-                            final IWorkbench workbench = PlatformUI.getWorkbench();
-                            final IWorkbenchWindow workbenchWindow = UIUtils.getActiveWorkbenchWindow();
-
-                            // Arm shutdown listener now because later will be too late
-                            final IWorkbenchListener listener = new IWorkbenchListener() {
-                                {
-                                    workbench.addWorkbenchListener(this);
-                                }
-
-                                @Override
-                                public boolean preShutdown(IWorkbench workbench, boolean forced) {
-                                    return true;
-                                }
-
-                                @Override
-                                public void postShutdown(IWorkbench workbench) {
-                                    try {
-                                        installer.run(file, log);
-                                    } catch (Exception e) {
-                                        log.error("Failed to run the installer script", e);
-                                    }
-                                }
-                            };
-
-                            UIUtils.asyncExec(() -> {
-                                ActionUtils.runCommand(IWorkbenchCommandConstants.FILE_EXIT, workbenchWindow);
-
-                                if (!workbench.isClosing()) {
-                                    workbench.removeWorkbenchListener(listener);
-                                    ShellUtils.launchProgram(folder.toString());
-                                }
-                            });
-                        } else {
-                            ShellUtils.showInSystemExplorer(file.toAbsolutePath().toString());
-                        }
+                UIUtils.asyncExec(() -> {
+                    workbench.close();
```

Comment:
So, the fix essentially involves using the `workbench.close()` method rather than invoking the `IWorkbenchCommandConstants.FILE_EXIT` command?

I'm not sure how the original approach caused two confirmations to show up. Can you elaborate, please?